scripts/security/scan.py

```diff
@@ -0,0 +1,139 @@
+    token = fd.readline().rstrip()
+
+def query(text, field, params={}):
+    result = []
+    start = 1
+
+    while True:
+        params["startAt"] = start
+        r = requests.get(baseurl + text, auth=auth, params=params)
+        if r.status_code != 200:
+            print(r)
+            raise Exception("Failure in query")
+        j = r.json()
+        # print(len(j[field]), j["maxResults"])
+
+        # The Jira API is inconsistent.  If the results returned are
+        # directly a list, just use that.
+        if isinstance(j, list):
+            return j
+        # for k in j.keys():
+        #     print(k)
+
+        result.extend(j[field])
+
+        if len(j[field]) < j["maxResults"]:
+            break
+
+        start += j["maxResults"]
+
+    return result
+
+def get_remote_links(key):
+    return query("issue/" + key + "/remotelink", 'unknown')
+
+class Issue(object):
+    def __init__(self, js):
+        self.key = js["key"]
+        fields = js["fields"]
+        self.fixversion = fields["fixVersions"]
+        self._status = fields["status"]
+        self._issuetype = fields["issuetype"]
+        self.cve = fields[cve_field]
+        self.embargo = fields[embargo_field]
+        self.subtasks = fields["subtasks"]
+
+        self.fields = fields
+
+        self.remotes = None
+        # Only query the remotes lazily, because it is slow.
+        # self.remotes = get_remote_links(self.key)
+
+    def status(self):
+        return self._status["name"]
+
+    def issuetype(self):
+        return self._issuetype["name"]
+
+    def getlinks(self):
+        if self.remotes is None:
+            self.remotes = get_remote_links(self.key)
+        return [x["object"]["url"] for x in self.remotes]
+
+def main():
+    p = { 'jql': 'project="ZEPSEC"' }
+    j = query("search", "issues", params=p)
+
+    pp = pprint.PrettyPrinter()
+    # j = r.json()
+
+    # pp.pprint(j['issues'][1])
+    issues = []
+    for jissue in j:
+        # print(jissue['key'])
+        issue = Issue(jissue)
+        issues.append(issue)
+
+    # Filter out the issues that are "Public".
+    issues = [x for x in issues if x.status() != "Public"]
+    issues = [x for x in issues if x.status() != "Rejected"]
+
+    # Select open primary issues (with CVE).
+    if False:
+        issues = [x for x in issues if x.issuetype() != "Backport"]
+        issues = [x for x in issues if x.cve is not None]
+
+    # Select unresolved backports
+    if True:
+        issues = [x for x in issues if x.issuetype() == "Backport"]
+
+    for issue in issues:
+        # print(issue.key, issue.cve)
+        print(issue.key, issue.status())
+        for link in issue.getlinks():
+            print("    ", link)
+            m = pr_re.search(link)
+            if m is not None:
+                pr = int(m.group(1))
+                gpr = repo.get_pull(pr)
+                print("        ", pr, gpr.state, gpr.title)
+                for c in gpr.get_commits():
```

Comment:
Why you iterating over all commits in the pr ? This won't work because the hash changes when rebased. So, if you want to check if it was merged we can just check it in the pr doing `gpr.merged`. If you want to know which branch it was merged, then we need to get the merge commit hash:
```
sha = gpr.merge_commit_sha
gitwork.describe(sha)
```

But we don't need to iterate over all commits. If you really need/want to to that I think you need to iterate the number of commits starting from `gpr.merge_commit_sha`, but I'm not seeing why we'd need that.